scrapy/extensions/postprocessing.py

```diff
@@ -0,0 +1,151 @@
+"""
+Extension for processing data before they are exported to feeds.
+"""
+from bz2 import BZ2File
+from gzip import GzipFile
+from io import IOBase
+from lzma import LZMAFile
+from typing import Any, BinaryIO, Dict, List
+
+from scrapy.utils.misc import load_object
+
+
+class GzipPlugin:
```

Comment:
Since all post-processing plugins have a common structure, how about declaring an interface (eg. `IPostProcessingPlugin`) with the `write` and `close` methods?

Reply:
While there is 1 case in similar Scrapy code where we use an interface, in most similar cases we tend to instead document the interface, so the current approach I think is the one most in line with the current code base.

In Python, interfaces are seldom used, and I don’t see much benefit for them being used here.